dw-snackbar.js

```diff
@@ -252,18 +249,29 @@ export class DwSnackbar extends layoutMixin(LitElement) {
      */
     this.constructor.counter = 0;
     this.positionVertical = "bottom";
+    this._onResize = this._debounceResize();
+  }
 
-    window.addEventListener("resize", this._getLayout.bind(this));
+  connectedCallback() {
+    super.connectedCallback();
+    window.addEventListener("resize", this._onResize);
   }
 
-  _getLayout() {
-    if (window.innerWidth > 768 && this.mobile) {
-      this.mobile = false;
-    }
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    window.removeEventListener("resize", this._onResize);
+  }
 
-    if (window.innerWidth < 768 && !this.mobile) {
-      this.mobile = true;
-    }
+  _debounceResize() {
```

Comment:
@devangdreamworld this isn't our standard practice to make any function debounced. Instead, it should be as follows:

1. Declare that function regularly e.g. `_onResise() {....}`
2. From the constructor make it debounced. e.g.
```javascript
this._onResize = debounce(this._onResize, 200);
```

In addition to this, when we bind an event imperatively `addEventListener()` and want to use `this` in the event listener function; and expect that `this` should be the current element. Then that function should be bound to `this` once from the custructor. e.g.

```javascript
this._onResize = this._onResize.bind(this);
```